lang/ruby30/Portfile

```diff
@@ -50,18 +50,59 @@ depends_skip_archcheck pkgconfig
 select.group        ruby
 select.file         ${filespath}/ruby30
 
-# patch-random.c.diff: fix build error at Lion (#63571)
-patchfiles          patch-random.c.diff
+# Patches derived from MacPorts-enhanced GitHub fork at
+# github.com/fhgwright/ruby
+
+# patch-sources.diff: fixes for build problems on 10.4 and 10.5.
+# Also, since the previous patch-random.c.diff (fix for 10.7 build error)
+# wasn't renormalized for 3.0.5, it has been folded into this diff, rather
+# than being retained as is.
+#
+# This diff is from v3_0_5 vs. macports-3_0_5r1.
+#
+patchfiles-append   patch-sources.diff
+
+# patch-generated.diff: additional patches to generated files, which don't
+# exist in the upstream git sources, and which must be patched after the
+# source patches so that make doesn't regard them as outdated.
+# At present, the only patched generated file is 'configure', to avoid
+# the need to rerun autoconf, whose output has reproducibility issues.
+# This diff is from tarball-3_0_5 vs. macports-tarball-3_0_5r1.
+#
+patchfiles-append   patch-generated.diff
+
+# It's recommended that the reinplace patch be folded into the combined
+# source patch in a future update (the fork already has a commit for it).
+# Using a patch instead of reinplace for the 'gem' versioning fix
+# would be more readable and more trackable.  A commit version of this already
+# exists in the fork, in the macports-reinplace-3_0 branch.
 
-#----------------------------------------------------------------------------------------
+#-------------------------------------------------------------------------------
 # Fix compilation on buildbots for 10.12 and earlier.
 # Issue: 62183: error: use of undeclared identifier 'fmt'; did you mean 'fma'?
-#----------------------------------------------------------------------------------------
+#
+# This issue only appears in Xcode clang 9 (clang 900), not MacPorts clang 9
+# (clang 901), so the blacklisting can be narrow.
+#-------------------------------------------------------------------------------
 compiler.blacklist-append \
-                    { clang < 901 }
+                    { clang >= 900 < 901 }
 
 compiler.thread_local_storage yes
 
+# At least one version of clang doesn't offer cstdbool (introduced in C++11)
+# with -stdlib=libstc++ but still reports as C++14.  This causes an include
+# failure in include/ruby/internal/stdbool.h whel built as C++.  Setting the
```

Comment:
typo: `whel` -> `when`

Reply:
I guess it's two steps forward and one back in the comment typo department. :-)

The changes are easier to read in my fork than in the patchfiles.

I'll force-push twice - once for the rebase and once for the real change, since this PR is currently 578 commits behind.

Reply:
OK, the second forced push is the typo correction.